analyses/deep-review-contrib/install-manual-r-dependencies.r

```diff
@@ -1,6 +1,6 @@
 packages = c(
   'https://cran.r-project.org/src/contrib/svglite_1.2.1.tar.gz',
-  'https://cran.r-project.org/src/contrib/ggridges_0.4.1.tar.gz'
+  'https://cran.r-project.org/src/contrib/Archive/ggridges/ggridges_0.4.1.tar.gz'
```

Comment:
@kurtwheeler is there a URL structure for versioned R dependencies that supports both the latest version and archived versions? IIRC you had to deal with this at some point, and there was an undocumented workaround.

Reply:
There is not such a URL structure. Our solution to this problem is to use [devtools](https://cran.r-project.org/web/packages/devtools/index.html) to install packages rather than specifying URLs directly.

However if you want to use the URL directly [this file](https://github.com/AlexsLemonade/refinebio/blob/b82715bd10aa0d3dbac4455b6e944f337ffdb229/common/install_devtools.R) might be of interest to you.

Reply:
> There is not such a URL structure

But I remember there being some undocumented way to retrieve it, like with `?version=X.X.X` appended to the URL or something. However, `devtools::install_version` [doesn't seem to use](https://rdrr.io/cran/remotes/src/R/install-version.R) this method I'm thinking of.

Cross-referencing [this tweet](https://twitter.com/BradleyJEck/status/919925180012654592) (unresolved).

Reply:
It looks like `remotes::install_version` is more lightweight of a dependency than `devtools`.

Reply:
@kurtwheeler didn't you [get the answer at the mailing list here](http://r.789695.n4.nabble.com/Suggestion-Make-CRAN-source-URLs-immutable-tp4753344.html)? For example, the following URLs should work?

```
https://cran.r-project.org/package=svglite&version=1.2.1
https://cran.r-project.org/package=ggridges&version=0.4.1
```

Reply:
@agitter can you see if those two URLs work?

Reply:
Replied to a stackoverflow with the [undocumented method](https://stackoverflow.com/a/54991606/4651668).

Reply:
Testing this in 8aff589.  I also changed other parts of the build so I may need to stabilize those before I can tell whether this worked.

Reply:
I watched the build logs and these URLs appeared to work.  As expected, I still need to fix other parts of the build before I can test the packages are functional.

Reply:
Installing the R packages with these CRAN URLs worked.